File: Python Workbook/8. AVERAGE CALC[2].py
#AVERAGE CALC[2]
repeating = True

while repeating:
    counter = 0
    numbers = [0, 0, 0, 0, 0]
    while counter < 5:
        try:
            numbers[counter] = float(input("Enter a value: "))
            counter +=1
        except ValueError:
            print("Enter a numerical value")
        except:
            print("What the hell did you do??")
    
    #Work out the average
    numbersTotal = 0
    for i in numbers:
        numbersTotal += i

    numbersAverage = numbersTotal / 5
    print("The average is {}.".format(numbersAverage))

    #Works out the sum of the first two numbers and then subtracts a third
    weirdNumber = numbers[0] * numbers[1] - numbers[3]
    print("The sum of the first two numbers minus the third is {}.".format(weirdNumber))
    
    while True:
        userRepeat = input("Do you want to repeat? \nEnter Y/N: ").lower()
        if userRepeat == "y":
            break
        elif userRepeat == "n":
            repeating = False
            break
        else:
            print("Enter valid input.")
    

File: Python Workbook in C++/8. AVERAGE CALC [2]/8. AVERAGE CALC [2]/8. AVERAGE CALC [2].cpp
// 8. AVERAGE CALC [2].cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <iostream>
#include <string>

using namespace std;

int main()
{
	string userRepeat;
	

	bool repeating = true;
	while (repeating){
		bool numberValid = false;
		int numbers[5] = { 0, 0, 0, 0, 0 };
		int counter = 0;
		int total = 0;
		float average = 0; 
		float weirdNumber = 0;

		while (counter < 5){
			bool numberValid = false;
			do
			{
				cout << "Please enter an integer: ";
				cin >> numbers[counter];

				if (cin.good()) {
					numberValid = true;
					counter += 1;
				}
				else {
					cout << "Enter a numerical value. " << endl;
					//something went wrong, we reset the buffer's state to good
					cin.clear();
					//and empty it
					cin.ignore(numeric_limits<streamsize>::max(), '\n');
				}
			} while (!numberValid);
		}

		for (int i = 0; i < 5; i++){
			total += numbers[i];
		}

		average = total / 5.0;
		cout << "The average is " << average << endl;

		weirdNumber = numbers[0] * numbers[1] - numbers[3];
		cout << "The sum of the first two numbers minus the third is " << weirdNumber << endl; 

		// Allows the user to quit
		while (true) {
			cout << "Do you want to repeat." << endl << "Enter Y/N: ";
			cin >> userRepeat;

			if ((userRepeat == "y") || (userRepeat == "Y")) {
				break;
			}
			else if ((userRepeat == "n") || (userRepeat == "y")) {
				repeating = false;
				break;
			}
			else {
				cout << "Enter a valid input." << endl << endl;
			}
		}
	}

	return 0;
}

